### PR TITLE
[chore]: refactor spanmetricprocessor to use only one map

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: processor/spanmetricsprocessor/processor.go [hunk 1]
```diff
@@ -382,61 +375,44 @@ func (p *processorImp) aggregateMetricsForSpan(serviceName string, span ptrace.S
...
-	p.latencyBucketCounts = make(map[metricKey][]uint64)
+	p.histograms = make(map[metricKey]*histogramData)
 	p.metricKeyToDimensions.Purge()
 }
 
-// updateLatencyExemplars sets the histogram exemplars for the given metric key and append the exemplar data.
-func (p *processorImp) updateLatencyExemplars(key metricKey, value float64, traceID pcommon.TraceID, spanID pcommon.SpanID) {
-	if _, ok := p.latencyExemplarsData[key]; !ok {
-		p.latencyExemplarsData[key] = []exemplarData{}
+// updateHistogram sets the histogram exemplars for the given metric key and append the exemplar data.
```
Comment (albertteoh) [inline processor/spanmetricsprocessor/processor.go:391]: I think this comment needs to be updated since it's no longer specific to updating exemplars.

**Review by albertteoh — APPROVED** [review]
Nice elegant, simplification. Thanks!

File: processor/spanmetricsprocessor/processor.go (hunk 1 above)
Comment (bogdandrutu) [reply processor/spanmetricsprocessor/processor.go]: Done.

**Review by albertteoh — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**